cpp/src/arrow/ipc/writer.cc

```diff
@@ -830,7 +830,9 @@ class RecordBatchStreamWriter::RecordBatchStreamWriterImpl : public StreamBookKe
 
     // Write 0 EOS message
     const int32_t kEos = 0;
-    return Write(&kEos, sizeof(int32_t));
+    RETURN_NOT_OK(Write(&kEos, sizeof(int32_t)));
+    // Align the stream for further data to be written to it
+    return Align();
```

Comment:
I'm not sure if this could cause issues or not. We could add to the IPC specification that consecutive streams written to a contiguous memory region should have padding bytes written at the end so that the next stream starts on an aligned offset. Thoughts?

Reply:
I agree with you that this bears some more analysis. Since I'm the author of much of this code, I would like to take some time to look at this and see what's the best way to approach. There were some other changes around padding and alignment for Tensor messages that I wasn't sure about. 

It would be good to address the under-specification of the streaming format that you cite -- having the last message be 8 or 64 bytes would make things tidier than they are now.